lib/screens/family_members_add_screen.dart

```diff
@@ -0,0 +1,332 @@
+                      Navigator.pop(context);
+                    },
+                    style: ElevatedButton.styleFrom(
+                      primary: Colors.transparent,
+                      shadowColor: Colors.transparent,
+                    ),
+                    child: Text(
+                      AppLocalizations.of(context)!.abort,
+                      style: const TextStyle(
+                        color: Colors.white,
+                        fontWeight: FontWeight.bold,
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+            Expanded(
+              child: Padding(
+                padding: const EdgeInsets.all(5),
+                child: Container(
+                  height: 44,
+                  decoration: BoxDecoration(
+                    gradient: const LinearGradient(
+                      begin: Alignment.topCenter,
+                      end: Alignment.bottomCenter,
+                      colors: <Color>[
+                        Color.fromRGBO(149, 193, 31, 1),
+                        Color.fromRGBO(110, 171, 39, 1),
+                      ],
+                    ),
+                    borderRadius: BorderRadius.circular(5),
+                  ),
+                  child: ElevatedButton(
+                    onPressed: () {
+                      addFamilyMember(context, _formKey);
+                    },
+                    style: ElevatedButton.styleFrom(
+                      primary: Colors.transparent,
+                      shadowColor: Colors.transparent,
+                    ),
+                    child: Text(
+                      AppLocalizations.of(context)!.save,
+                      style: const TextStyle(
+                        color: Colors.white,
+                        fontWeight: FontWeight.bold,
+                      ),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// returns a message (String)
+/// whether e-mail-address validation is successful or not
+String? validateEmail(String value, BuildContext context) {
+  String pattern =
+      r"^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]"
+      r'{0,253}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]'
+      r'{0,253}[a-zA-Z0-9])?)*$';
+
+  RegExp regex = RegExp(
+    pattern,
+  );
+
+  if (!regex.hasMatch(value)) {
+    return AppLocalizations.of(context)!.entryValidEmail;
+  }
+
+  return null;
+}
+
+/// When 'save'-button is pressed, this method will be triggered.
+/// After successful validation, data will be written in database.
+void addFamilyMember(BuildContext context, GlobalKey<FormState> key) {
+  if (key.currentState!.validate()) {
+    ScaffoldMessenger.of(context).showSnackBar(
+      SnackBar(
+        content: Text(
+          AppLocalizations.of(context)!.submitData,
+        ),
+      ),
+    );
+  }
+}
```

Comment:
Is this code supposed to execute when pressing the "add" Button in the family members screen?
I'm getting an error when doing this. Is this an accounted behaviour?

Reply:
What kind of error do you get? Actually, it shouldn't do anything after pressing the "add" Button

Reply:
There is an onPressed function that uses a `Navigator.push()` to FamilyMembersAdd().
Doing this I'm getting a red screen with an assertion error.

Reply:
This problem should be fixed now!

Reply:
> I noticed that in some files (add_medication_screen/add_medication_screen_label.dart, add_medication_screen/add_medication_screen.dart, my_medications_screen/medication_card_tile.dart, my_medications_screen/medication_card.dart, my_medications_screen/medication_list.dart, my_medications_screen/my_medications_screen.dart) the GNU licence text is missing.
> 
> Isn't it mandatory in our project to put this licence text at the top of each file?

@S1rDan 

The GPL 3.0 has got a "how to apply" section at the bottom and it says the following:

> How to Apply These Terms to Your New Programs
> If you develop a new program, and you want it to be of the greatest possible use to the > public, the best way to achieve this is to make it free software which everyone can
> redistribute and change under these terms.

> To do so, attach the following notices to the program. It is safest to attach them to
> the start of each source file to most effectively state the exclusion of warranty; and
> each file should have at least the “copyright” line and a pointer to where the full
> notice is found.